Lib9c/Model/Item/ShopItem.cs

```diff
@@ -61,6 +82,10 @@ public ShopItem(Dictionary serialized)
             Costume = serialized.ContainsKey("costume")
                 ? (Costume) ItemFactory.Deserialize((Dictionary) serialized["costume"])
                 : null;
+            if (serialized.ContainsKey(ExpiredBlockIndexKey))
+            {
+                ExpiredBlockIndex = serialized[ExpiredBlockIndexKey].ToLong();
+            }
```

Comment:
As extension of [`ExpiredBlockIndex` negative check comment](https://github.com/planetarium/lib9c/pull/317/files#r585278378), it should check it also here to resist invalid value. Of course, I couldn't see using shop item as argument in plain value of an action (i.e. the situation to allow users to create shop item manually). But I believe it will improve `ShopItem` class more safely.

If you agree with my opinion, how about checking it also here and using `ExpiredBlockIndex` as property and share the logic to check through `setter`?